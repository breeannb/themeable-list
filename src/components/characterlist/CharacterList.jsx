import React from 'react';
import PropTypes from 'prop-types';

const CharacterList = ({ characters }) => {


  const listElement = characters.map((character, i) => {
    return (
      <li key={i} >
        <img src={character.photoUrl} />
        <p>{character.name}</p>
        <p>{character.allies}</p>
        <p>{character.enemies}</p>
        <p>{character.affiliations}</p>
      </li>
    );
  });

  return (
    <ul>
      {listElement}
    </ul>
  );

};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.object.isRequired).isRequired
};

export default CharacterList;
