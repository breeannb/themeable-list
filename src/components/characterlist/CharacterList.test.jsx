import React from 'react';
import { shallow } from 'enzyme';
import CharacterList from './CharacterList';

describe('CharacterList component', () => {
  it('renders CharacterList', () => {
    const wrapper = shallow(<CharacterList 
      characters={[{  
        name: 'Aang', 
        photoUrl: 'image1.jpg', 
        allies: 'Sokka' 
      }, 
      { name: 'Aang2', 
        photoUrl: 'image2.jpg', 
        allies: 'Sokka2'  }]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
