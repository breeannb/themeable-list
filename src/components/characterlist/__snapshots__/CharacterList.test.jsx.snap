// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<ul>
  <li
    key="0"
  >
    <img
      src="image1.jpg"
    />
    <p>
      Aang
    </p>
    <p>
      Sokka
    </p>
    <p />
    <p />
  </li>
  <li
    key="1"
  >
    <img
      src="image2.jpg"
    />
    <p>
      Aang2
    </p>
    <p>
      Sokka2
    </p>
    <p />
    <p />
  </li>
</ul>
`;
